signer/src/storage/postgres.rs

```diff
@@ -1376,18 +1376,21 @@ impl super::DbRead for PgStore {
         chain_tip: &model::BitcoinBlockHash,
         context_window: u16,
     ) -> Result<Vec<model::SweptDepositRequest>, Error> {
-        // TODO: This query needs to be updated to check that the
-        // `completed_deposit_event` is in a Stacks block linked to the
-        // canonical Bitcoin chain (i.e. confirmed) once #559 is completed.
-
         // The following tests define the criteria for this query:
         // - [X] get_swept_deposit_requests_returns_swept_deposit_requests
         // - [X] get_swept_deposit_requests_does_not_return_unswept_deposit_requests
-        // - [ ] get_swept_deposit_requests_does_not_return_deposit_requests_with_responses (needs #559)
-        // - [ ] get_swept_deposit_requests_response_tx_reorged (needs #559)
+        // - [X] get_swept_deposit_requests_does_not_return_deposit_requests_with_responses
+        // - [X] get_swept_deposit_requests_response_tx_reorged
+
+        let Some(stacks_chain_tip) = self.get_stacks_chain_tip(chain_tip).await? else {
+            return Ok(Vec::new());
+        };
 
         sqlx::query_as::<_, model::SweptDepositRequest>(
             "
+            WITH s_blocks AS (
+                SELECT * FROM stacks_blockchain_of($3, $1, $2)
+            )
```

Comment:
Can we include this in the left join below? Like so?
```sql
LEFT JOIN stacks_blockchain_of($3, $1, $2) AS sb
  ON ...
```

Reply:
[c3e0400](https://github.com/stacks-network/sbtc/pull/747/commits/c3e04003d7fb9578c919756d7ed254180c3df45a)